avocado-setup.py

```diff
@@ -446,20 +448,23 @@ def edit_mux_file(test_config_name, mux_file_path, tmp_mux_path):
     with open(mux_file_path) as mux_fp:
         mux_str = mux_fp.read()
 
-    mux_str_edited = []
-    for line in mux_str.splitlines():
-        if len(line) == 0 or line.lstrip()[0] == '#':
-            continue
-        for key, value in input_dic.items():
-            temp_line = line.split(":")
-            mux_key = temp_line[0]
-            mux_value = temp_line[1]
-            if key == mux_key.strip():
-                line = line.replace('%s' % line.strip(), '%s: %s' % (key, value))
-        mux_str_edited.append(line)
+    yaml.constructor.SafeConstructor.add_constructor(
+        None, yaml_helper.OverrideConstructor.construct_undefined)
+    mux_json = yaml.load(mux_str, yaml_helper.OverrideLoader)
 
+    for key, value in input_dic.items():
+        if helper.keys_exists(mux_json, *(key.split("."))):
+            if value.lower() in ['true', 'false']:
+                helper.deep_put(
+                    key.split("."), mux_json, json.loads(value.lower()))
```

Comment:
@harish-24 Is this to keep `true` as `true` ?

Reply:
yes, we read the configuration as strings and this makes sure to write it back as bool. Another option would be to read bool from config, but since the existing just stores as strings, handled the same way.